rel/expr_binary.go

```diff
@@ -306,7 +307,28 @@ func evalValForAddArrow(lhs, rhs Value) (Value, error) {
 		}
 	}
 
+	if lhs, ok := lhs.(Dict); ok {
+		if rhs, ok := rhs.(Dict); ok {
+			return mergeDicts(lhs, rhs), nil
+		}
+	}
+
 	return nil, errors.Errorf(
-		"Both args to %q must be tuples, not %T and %T",
+		"Both args to %q must be both tuples or both dicts, not %T and %T",
 		"+>", lhs, rhs)
 }
+
+func mergeDicts(lhs Dict, rhs Dict) Dict {
+	tempMap := lhs.m
+
+	for e := rhs.DictEnumerator(); e.MoveNext(); {
+		key, value := e.Current()
+
+		if lhs.m.Has(key) {
+			tempMap = tempMap.Without(frozen.NewSet(key))
+		}
+		tempMap = tempMap.With(key, value)
```

Comment:
I don't think you need to do a `Without` and then a `With` operation. Just `With` should do it, does it not work?